include/commissioner/commissioner.hpp

```diff
@@ -1147,6 +1161,36 @@ class Commissioner
      *         git repository.
      */
     static std::string GetVersion(void);
+
+    /**
+     * @brief Asynchronously query diagnostic decoded data from a Thread device.
+     *
+     * This method sends a DIAG_GET.qry message to the specified Thread device,
+     * requesting the set of diagnostic data indicated by `aDiagDataFlags`.
+     * The response, or any errors encountered, will be delivered to the provided `aHandler`.
+     *
+     * @param[in, out] aHandler        A handler to process the response or any errors.
+     *                                 This handler is guaranteed to be called.
+     * @param[in]      aAddr           Mesh local address of the target Thread device.
```

Comment:
Does this have to be the Mesh-local address? Can link-local or OMR address work?

Reply:
Based on spec 8.6.2, the diagnostic messages exchange between  commissioner and thread devices is implemented by proxy commands (UDP_RX and UDP_TX), and the IPv6 address should be mesh local address.